### PR TITLE
Fix failure when ACID table column name collides with ACID internal names

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Fix failure when building column metadata in Hive

## Documentation

(x) No documentation is needed.

## Release notes

(x) Release notes entries required with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Comment by ebyhr** [comment]
CI hit #12405

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java [hunk 1]
```diff
@@ -3411,29 +3405,12 @@ private static Function<HiveColumnHandle, ColumnMetadata> columnMetadataGetter(T
             }
         }
 
-        // add hidden columns
-        builder.put(PATH_COLUMN_NAME, Optional.empty());
```
Comment (findepi) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java:3415]: why removed?

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java [hunk 2]
```diff
@@ -3411,29 +3405,12 @@ private static Function<HiveColumnHandle, ColumnMetadata> columnMetadataGetter(T
...
-            }
-        }
-
         Map<String, Optional<String>> columnComment = builder.buildOrThrow();
 
         return handle -> ColumnMetadata.builder()
                 .setName(handle.getName())
                 .setType(handle.getType())
-                .setComment(columnComment.get(handle.getName()))
+                .setComment(handle.isHidden() ? Optional.empty() : columnComment.get(handle.getName()))
```
Comment (findepi) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java:3413]: why change?

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseHiveConnectorTest.java [hunk 3]
```diff
@@ -8328,6 +8328,30 @@ public void testUseColumnNames(HiveStorageFormat format, boolean formatUseColumn
...
 
+    @Test(dataProvider = "hiddenColumnNames")
+    public void testHiddenColumnNameConflict(String columnName)
+    {
+        try (TestTable table = new TestTable(
+                getQueryRunner()::execute,
+                "test_hidden_column_name_conflict",
+                format("(\"%s\" int, _bucket int, _partition int) WITH (partitioned_by = ARRAY['_partition'], bucketed_by = ARRAY['_bucket'], bucket_count = 10)", columnName))) {
+            assertThatThrownBy(() -> query("SELECT * FROM " + table.getName()))
+                    .hasMessageContaining("Multiple entries with same key: " + columnName);
```
Comment (findepi) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseHiveConnectorTest.java:8339]: We would probably want disallow creation of such tables

**Comment by findepi** [comment]
What problem(s) is this fixing?

**Comment by ebyhr** [comment]
@findepi The below `buildOrThrow()` in `HiveMetadata.columnMetadataGetter()` threw an exception when the table contains column names in `AcidSchema.ACID_COLUMN_NAMES` (e.g. `operation`) and it's the ACID table. 
```java
        Map<String, Optional<String>> columnComment = builder.buildOrThrow();
```

**Comment by findepi** [comment]
> @findepi The below `buildOrThrow()` in `HiveMetadata.columnMetadataGetter()` threw an exception when the table contains column names in `AcidSchema.ACID_COLUMN_NAMES` (e.g. `operation`) and it's the ACID table.
> 
> ```java
>         Map<String, Optional<String>> columnComment = builder.buildOrThrow();
> ```

Thanks for explanation. This makes this change clear to me now.

Please add mention of that in the commit message

**Review by findepi — APPROVED** [review]

**Comment by findepi** [comment]
pushed; updated the commit title only.

**Comment by ebyhr** [comment]
Thanks, removed my old commit title. 
Screen Shot 2022-05-18 at 8 52 42

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseHiveConnectorTest.java (hunk 3 above)
Comment (ebyhr) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseHiveConnectorTest.java:8339]: Filed #12449

**Pull request merged**